src/test/java/com/jcabi/s3/AwsBucketTest.java

```diff
@@ -106,4 +115,121 @@ public void existsThrowsIOException() throws IOException {
         bucket.exists();
     }
 
+    /**
+     * AwsBucket supports listing large buckets.
+     * @throws Exception if test fails
+     */
+    @Test
+    public void supportsListingLargeBuckets() throws Exception {
+        final ListingLargeBucketsTestData data =
+            new ListingLargeBucketsTestData();
+        final Region region = this.mockRegionForListingLargeBuckets(data);
+        final Bucket bucket = new AwsBucket(region, data.bucketName());
+        final Iterator<String> actual = bucket.list(data.prefix()).iterator();
+        Assert.assertTrue(actual.hasNext());
+        Assert.assertEquals(data.firstItem(), actual.next());
+        Assert.assertTrue(actual.hasNext());
+        Assert.assertEquals(data.secondItem(), actual.next());
+        Assert.assertFalse(actual.hasNext());
+    }
+
+    /**
+     * Returns a mocked region for supportsListingLargeBuckets test.
+     * @param data Test data
+     * @return Mocked region
+     */
+    private Region mockRegionForListingLargeBuckets(
+        final ListingLargeBucketsTestData data
+    ) {
+        final Region region = Mockito.mock(Region.class);
```

Comment:
Actually, MKRegion is not yet implemented, so mocking is the only option I have to emulate AWS behavior in order to test the method.